data/user_data_base.ps2

```diff
@@ -199,13 +210,28 @@ function Get-MyName {
     return $nameTag.Value
 }
 
+$timeout = new-timespan -Minutes 5
+$sw = [diagnostics.stopwatch]::StartNew()
+
+### Wait until tags are set or it's longer than 5 minutes
+Do {
+    Write-Output "Waiting for tags to be added"
+} while ((-not (Check-AnyTagExists) ) -and ($sw.elapsed -lt $timeout))
+
+if ($sw.elapsed -ge $timeout) {
```

Comment:
Instead of checking if the elapsed time is greater than the timeout, would it make sense to just check the value of `Check-AnyTagExists`? Although unlikely, it'd be possible for us to exit the loop above at something like 4 minutes and 59 seconds (assuming the `-and` becomes an `-or` and the tags get set at that point) and then check if the elapsed time is greater than the timeout immediately after the 5 minute mark. 

So we'd have exited the loop because the tags got set, but the log message would say that it timeout while waiting for the tags to get set.